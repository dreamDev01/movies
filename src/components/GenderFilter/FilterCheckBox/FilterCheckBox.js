import { useState } from "react";

import "./FilterCheckBox.scss";

/*-------Filter Checkbox----------*/
/*props :        */
/*  - name       */
/*  - label      */
/*  - count      */
/*  - onSelect   */
/*  - onUnSelect */

const FilterCheckBox = (props) => {
  const { name, label, count, onSelect, onUnSelect } = props;
  const [isChecked, setIsChecked] = useState(false);

  const handleChange = (e) => {
    const name = e.target.name;

    setIsChecked(!isChecked);
    if (e.target.checked) {
      onSelect(name);
    } else {
      onUnSelect(name);
    }
  };

  return (
    <div className="checkbox">
      <input
        data-testid={name}
        name={name}
        type="checkbox"
        checked={isChecked}
        onChange={handleChange}
      ></input>
      {label} ({count.toString()})
    </div>
  );
};

export default FilterCheckBox;
