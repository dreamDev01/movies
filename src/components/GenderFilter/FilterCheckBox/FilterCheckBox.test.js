import renderer, { act } from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import FilterCheckBox from "./FilterCheckBox";

describe("tests FilterCheckBox", () => {
  const props = {
    name: "Drama",
    label: "Dramatique",
    count: 2,
    onSelect: jest.fn(),
    onUnSelect: jest.fn(),
  };
  it("should render component", async () => {
    // Arrange
    const tree = renderer.create(<FilterCheckBox {...props} />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  describe("display", () => {
    it("should display filter checkbox", async () => {
      // Act
      await act(async () => {
        render(<FilterCheckBox {...props} />);
      });

      // Assert
      expect(
        screen.getByText(props.label + " (" + props.count + ")")
      ).toBeInTheDocument();
      expect(screen.getByTestId(props.name).checked.toString()).toEqual(
        "false"
      );
    });
  });
});
