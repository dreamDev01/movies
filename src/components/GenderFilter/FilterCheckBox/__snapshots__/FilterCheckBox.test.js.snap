// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests FilterCheckBox should render component 1`] = `
<div
  className="checkbox"
>
  <input
    checked={false}
    data-testid="Drama"
    name="Drama"
    onChange={[Function]}
    type="checkbox"
  />
  Dramatique
   (
  2
  )
</div>
`;
